graph/pull_v2.go

```diff
@@ -226,18 +249,32 @@ func (p *v2Puller) pullV2Tag(out io.Writer, tag, taggedName string) (verified bo
 			out.Write(p.sf.FormatProgress(stringid.TruncateID(img.ID), "Already exists", nil))
 			continue
 		}
-		out.Write(p.sf.FormatProgress(stringid.TruncateID(img.ID), "Pulling fs layer", nil))
 
+		digest := manifest.FSLayers[i].BlobSum
 		d := &downloadInfo{
 			img:     img,
 			poolKey: "layer:" + img.ID,
-			digest:  manifest.FSLayers[i].BlobSum,
+			digest:  digest,
+			size:    0,
 			// TODO: seems like this chan buffer solved hanging problem in go1.5,
 			// this can indicate some deeper problem that somehow we never take
 			// error from channel in loop below
 			err: make(chan error, 1),
 		}
 
+		if dryRun {
+			size, err := p.extractSize(d)
+			if err != nil {
+				return false, err
+			}
+			d.size = size
```

Comment:
Seems like it's not necessary to set `d.size`, because it's not going to be used in the dry-run mode.